app/src/main/java/com/rockthevote/grommet/ui/MainActivityViewModel.kt

```diff
@@ -3,40 +3,46 @@ package com.rockthevote.grommet.ui
 import androidx.lifecycle.*
 import com.rockthevote.grommet.data.api.RockyService
 import com.rockthevote.grommet.data.api.model.RegistrationResponse
-import com.rockthevote.grommet.data.db.model.RockyRequest
+import com.rockthevote.grommet.data.db.dao.RegistrationDao
+import com.rockthevote.grommet.data.db.model.Registration
 import com.rockthevote.grommet.util.coroutines.DispatcherProvider
 import com.rockthevote.grommet.util.coroutines.DispatcherProviderImpl
 import kotlinx.coroutines.*
-import retrofit2.adapter.rxjava.Result
-import rx.Observable
+import okhttp3.MediaType
+import okhttp3.RequestBody
+import retrofit2.Response
 import timber.log.Timber
 
 class MainActivityViewModel(
-    dispatchers: DispatcherProvider = DispatcherProviderImpl(),
-    private val rockyService: RockyService
+    private val dispatchers: DispatcherProvider = DispatcherProviderImpl(),
+    private val rockyService: RockyService,
+    private val registrationDao: RegistrationDao
 ) : ViewModel() {
 
     private val _state = MutableLiveData<MainActivityState>(MainActivityState.Init)
     val state: LiveData<MainActivityState> = _state
 
     private val coroutineExceptionHandler = CoroutineExceptionHandler { _, throwable ->
-        // TODO Should we handle this another way? Is it needed?
         Timber.e(throwable)
     }
 
+    init {
+        refreshPendingUploads()
+    }
+
     private val supervisorJob = SupervisorJob()
 
-    private val rockyRequestScope = CoroutineScope(dispatchers.io + coroutineExceptionHandler)
+    private val rockyRequestScope = CoroutineScope(dispatchers.io + coroutineExceptionHandler + supervisorJob)
 
-    init {
+    fun refreshPendingUploads() {
         viewModelScope.launch(dispatchers.io) {
             updateState(MainActivityState.Loading)
 
             val requests = loadRequestsFromDb()
 
             val content = MainActivityState.Content(
                 pendingUploads = requests.size,
-                failedUploads = 0 // TODO need to figure out how we're going to calculate this
+                failedUploads = 0 // TODO need to figure out how/if we're going to calculate this
```

Comment:
Originally, I kept a status column in the registration table `pending,succeeded,failed`. Another option is a retry count in the database

Reply:
I think a retry count is probably the way to go here then. Uploaded registrations are removed from the DB, so no need for `succeeded`, then it becomes binary where `0 == pending` and `> 0 == failed` on the retry count.

Reply:
I added a `uploadAttempts` to the `Registration` object in the DB, but still not sure how this should reflect as a `failedUpload`, at this point in the code, all `pendingUploads` are technically failed uploads. 

We'll probably have to wait on the determination on how to handle errors before we go further on this, but at least for the time being, the `uploadAttempts` are being updated on each failure for each request.